File: Python/DSA/Searching/binarysearch.py
def binary_search(arr, x):
	low = 0
	high = len(arr) - 1
	mid = 0

	while low <= high:

		mid = (high + low) // 2

		if arr[mid] < x:
			low = mid + 1

		elif arr[mid] > x:
			high = mid - 1

		else:
			return mid

	return -1


arr = [ 8, 12, 54, 98, 223, 900 ]
x = 12

# Function call
result = binary_search(arr, x)

if result != -1:
	print("Element is present")
else:
	print("Element is not present")

File: C++/Data_Structure_And_Algorithms/Searching/Binary_Search.cpp
#include<bits/stdc++.h>
using namespace std;

int b_search(int arr[],int n,int key){
    int low=0,high=n-1;
    while(low<=high){
        int mid=(low+high)/2;
        if(arr[mid]==key)
            return mid;
        else if(arr[mid]>key)
            high=mid-1;
        else
            low=mid+1;
    }
    return -1;
}

int main(){
    int n;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++)
        cin>>arr[i];
    int key;
    cin>>key;
    cout<<b_search(arr,n,key);
}
